### PR TITLE
Log difference between old and new partition infos in PartitionDiscoveryThread, minor improvements for other logs

### PR DESCRIPTION
**Pull request by somandal**
Important: DO NOT REPORT SECURITY ISSUES DIRECTLY ON GITHUB.  
For reporting security issues and contributing security fixes,  
please, email security@linkedin.com instead, as described in  
the contribution guidelines.

Please, take a minute to review the contribution guidelines at:  
https://github.com/linkedin/Brooklin/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by vishwajith-s — DISMISSED** [review]

**Review by vishwajith-s — APPROVED** [review]

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java [hunk 1]
```diff
@@ -313,12 +313,23 @@ public void run() {
...
+
+            List<String> addedTopicPartitions = newPartitionInfo
+                .stream()
+                .filter(topicPartition -> !_subscribedPartitions.contains(topicPartition))
+                .collect(Collectors.toList());
+            List<String> removedTopicPartitions = _subscribedPartitions
+                .stream()
+                .filter(topicPartition -> !newPartitionInfo.contains(topicPartition))
+                .collect(Collectors.toList());
+            LOG.info("TopicPartitions for {} that are to be added: {} and that are to be removed: {}",
```
Comment (vmaheshw) [inline datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java:331]: IMO, Splitting this in two lines will make the search easier while debugging.

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java [hunk 2]
```diff
@@ -313,12 +313,23 @@ public void run() {
...
 
           if (!ListUtils.isEqualList(newPartitionInfo, _subscribedPartitions)) {
-            LOG.info("Get updated partition info for {}, oldPartitionInfo: {}, new Partition info: {}"
-                , datastream.getName(), _subscribedPartitions, newPartitionInfo);
+            LOG.info("Get updated Partitions Info for {}, old Partitions Info: {}, new Partitions Info: {}",
+                datastream.getName(), _subscribedPartitions, newPartitionInfo);
+
+            List<String> addedTopicPartitions = newPartitionInfo
+                .stream()
+                .filter(topicPartition -> !_subscribedPartitions.contains(topicPartition))
```
Comment (vmaheshw) [inline datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java:325]: Converting these list into set, cache the set as well and doing:

added = NewSet - ExistingSet or NewSet.removeall(existingSet)
removed = ExistingSet - NewSet or ExistingSet.removeall(NewSet)

will make this operation faster. Right now its O(N^2) complexity.

**Review by vmaheshw — DISMISSED** [review]

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java (hunk 1 above)
Comment (somandal) [reply datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java]: done

File: datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java (hunk 2 above)
Comment (somandal) [reply datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/mirrormaker/KafkaMirrorMakerConnector.java]: Modified to use a set, found the intersection of the two sets and removed the intersection from each.

**Review by vmaheshw — APPROVED** [review]
LGTM

**Review by vishwajith-s — APPROVED** [review]

**Pull request merged**